code/modules/power/power_store.dm

```diff
@@ -57,6 +57,13 @@
 	)
 	AddElement(/datum/element/connect_loc, loc_connections)
 
+
+/obj/item/stock_parts/power_store/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
+	. = ..()
+	if(!isobj(loc))
```

Comment:
this will call update_appearance the cell is moved via pulling or a conveyor or anything else
if you want it to only call it if the objects old location was not a turf (object or something else) then just do this
```suggestion
	if(!isturf(old_loc))
```

Reply:
This would work as well